source/js/plugins/metadata.js

```diff
@@ -34,21 +36,25 @@ export default class MetadataPlugin
      * Display a modal with the IIIF manifest metadata. Here, viewer refers to the Diva instance (because
      * the toolbar class refers to it as viewer)
      **/
-    handleClick (viewer) 
+    handleClick () 
     {
         // if first click create div elements
         let metadataDiv;
+
```

Comment:
Do you need to handle the absence of metadata here? If the toolbar icon is not created, then the click handler is never attached, thus this method will never get called.